Libraries/CommonFunctions.psm1

```diff
@@ -2437,6 +2437,45 @@ Function Restart-VMFromShell($VMData, [switch]$SkipRestartCheck) {
     }
 }
 
+Function Wait-AzVMBackRunningWithTimeOut($AllVMData, [scriptblock]$AzVMScript) {
+    if (!$AllVMData -or !$AllVMData.InstanceSize -or !$AllVMData.ResourceGroupName -or !$AllVMData.RoleName) {
+        return $false
+    }
+    $VMCoresArray = @()
+    $AzureVMSizeInfo = Get-AzVMSize -Location $AllVMData[0].Location
+    foreach ($vmData in $AllVMData) {
+        $AzVMScript.Invoke($vmData)
+        if (-not $?) {
+            Write-LogErr "Failed in AzVM operation for $($vmData.RoleName)"
+            return $false
+        }
+        $VMCoresArray += ($AzureVMSizeInfo | Where-Object { $_.Name -eq $vmData.InstanceSize }).NumberOfCores
+    }
+    $MaximumCores = ($VMCoresArray | Measure-Object -Maximum).Maximum
+
+    # Calculate timeout depending on VM size.
+    # We're adding timeout of 10 minutes (default timeout) + 1 minute/10 cores (additional timeout).
+    # So For D64 VM, timeout = 10 + int[64/10] = 16 minutes.
+    # M128 VM, timeout = 10 + int[128/10] = 23 minutes.
+    $Timeout = New-Timespan -Minutes ([int]($MaximumCores / 10) + 10)
```

Comment:
Suggest to print out this Timeout value in the test log. It's helpful.

Reply:
updated